src/postgraphile/http/subscriptions.ts

```diff
@@ -186,134 +184,278 @@ export async function enhanceHttpServerWithSubscriptions<
     options,
   });
 
-  SubscriptionServer.create(
-    {
-      schema,
-      validationRules: staticValidationRules,
-      execute: () => {
-        throw new Error('Only subscriptions are allowed over websocket transport');
-      },
-      subscribe: options.live ? liveSubscribe : graphqlSubscribe,
-      onConnect(
-        connectionParams: Record<string, any>,
-        _socket: WebSocket,
-        connectionContext: ConnectionContext,
-      ) {
-        const { socket, request } = connectionContext;
-        socket['postgraphileId'] = ++socketId;
-        if (!request) {
-          throw new Error('No request!');
-        }
-        const normalizedConnectionParams = lowerCaseKeys(connectionParams);
-        request['connectionParams'] = connectionParams;
-        request['normalizedConnectionParams'] = normalizedConnectionParams;
-        socket['__postgraphileReq'] = request;
-        if (!request.headers.authorization && normalizedConnectionParams['authorization']) {
+  switch (websockets) {
+    case 'none':
+      break;
+    case 'v0':
+      SubscriptionServer.create(
+        {
+          schema,
+          validationRules: staticValidationRules,
+          execute:
+            operations === 'all'
+              ? execute
+              : () => {
+                  throw new Error('Only subscriptions are allowed over websocket transport');
+                },
+          subscribe: options.live
+            ? (
+                schema,
+                document,
+                rootValue,
+                contextValue,
+                variableValues,
+                operationName,
+                fieldResolver,
+              ) =>
+                liveSubscribe({
+                  schema,
+                  document,
+                  rootValue,
+                  contextValue,
+                  variableValues,
+                  operationName,
+                  fieldResolver,
+                })
```

Comment:
Ah; you've moved the logic here from before. Please put it back where it was for parity with https://github.com/graphql/graphql-js/blob/master/src/execution/execute.js